### PR TITLE
Refactor Validate

### PR DESCRIPTION
**Pull request by beckje01**
This will refactor how validate work so we can seperate out invalid vs errored tokens checks.

### CODE REVIEW

File: ratpack-bearer-auth/src/main/java/st/ratpack/auth/ValidateTokenResult.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
+package st.ratpack.auth;
+
+public interface ValidateTokenResult {
+
+	Status getStatus();
+
+	OAuthToken getOAuthToken();
+
+	default Boolean isErrorResult() {
+		return this.getStatus().equals(Status.UNKNOWN);
```
Comment (jeff-blaisdell) [inline ratpack-bearer-auth/src/main/java/st/ratpack/auth/ValidateTokenResult.java:10]: Could we switch the order of these to ensure no NPEs?  
i.e.
```
return Status.UNKNOWN.equals(this.getStatus())
```

**Review by rappleg — APPROVED** [review]

**Review by mcacek — APPROVED** [review]

**Review by llinder — APPROVED** [review]
Looks good

File: ratpack-bearer-auth/src/main/java/st/ratpack/auth/ValidateTokenResult.java (hunk 1 above)
Comment (llinder) [reply ratpack-bearer-auth/src/main/java/st/ratpack/auth/ValidateTokenResult.java:10]: ha yeah good catch!

**Review by jeff-blaisdell — APPROVED** [review]

**Review by mrobinet — APPROVED** [review]

**Comment by mrobinet** [comment]
Merging this to get some movement. I'll do a followup PR to address Jeff's comment.

**Pull request merged**